2_process/src/process_data.R

```diff
@@ -0,0 +1,56 @@
+#this function will take the input data:
+  # 1) process these data
+  # 2) save these data
+
+process_data <- function(data){
+  
+  #process data to prepare for plotting 
+  eval_data <- data %>% dplyr::filter(stringr::str_detect(exper_id, 'similar_[0-9]+')) %>%
+    dplyr::mutate(col = case_when(
+      model_type == 'pb' ~ '#1b9e77',
+      model_type == 'dl' ~'#d95f02',
+      model_type == 'pgdl' ~ '#7570b3'
+    ), pch = case_when(
+      model_type == 'pb' ~ 21,
+      model_type == 'dl' ~ 22,
+      model_type == 'pgdl' ~ 23
+    ), n_prof = as.numeric(stringr::str_extract(exper_id, '[0-9]+')))
+  
+  
+  out_file_path <- here::here("2_process/out/","processed_data.csv")
+  
+  readr::write_csv(eval_data, 
+                    file = out_file_path)
```

Comment:
Currently, this function essentially has two outputs - the file and the actual data as an R object. It is best if each function just had one output. I would recommend moving the `write_csv()` step to its own function.

Reply:
Awesome, thanks for the feedback!